python/cuml/ensemble/randomforestregressor.pyx

```diff
@@ -620,40 +621,42 @@ class RandomForestRegressor(Base):
             ndarray, cuda array interface compliant array like CuPy
-        predict_model : String
-                        "GPU" if prediction should be carried out on the GPU
-                        "CPU" or None if prediction should be carried out
-                        on the CPU
-        output_class: boolean
-                      This is optional and required only while performing the
-                      predict operation on the GPU.
-                      If true, return a 1 or 0 depending on whether the raw
-                      prediction exceeds the threshold. If False, just return
-                      the raw prediction.
-        algo : string name of the algo from (from algo_t enum)
-               This is optional and required only while performing the
-               predict operation on the GPU.
-               'NAIVE' - simple inference using shared memory
-               'TREE_REORG' - similar to naive but trees rearranged to be more
-                              coalescing-friendly
-               'BATCH_TREE_REORG' - similar to TREE_REORG but predicting
-                                    multiple rows per thread block
+        predict_model : String (default = 'GPU')
+            'GPU' to predict using the GPU, 'CPU' otherwise. The GPU can only
+            be used if the model was trained on float32 data and `X` is float32
+            or convert_dtype is set to True.
+        algo : string (default = 'BATCH_TREE_REORG')
+            This is optional and required only while performing the
+            predict operation on the GPU.
+            'NAIVE' - simple inference using shared memory
+            'TREE_REORG' - similar to naive but trees rearranged to be more
+            coalescing-friendly
+            'BATCH_TREE_REORG' - similar to TREE_REORG but predicting
+            multiple rows per thread block
+        convert_dtype : bool, optional (default = True)
+            When set to True, the predict method will, when necessary, convert
+            the input to the data type which was used to train the model. This
+            will increase memory used for the method.
         Returns
         ----------
         y: NumPy
            Dense vector (int) of shape (n_samples, 1)
         """
-        if self.dtype == np.float64:
-            raise TypeError("GPU predict model only accepts float32 dtype"
-                            " as input, convert the data to float32 or "
-                            "use the CPU predict with `predict_model='CPU'`.")
-
-        elif predict_model == "GPU":
-            preds = self._predict_model_on_gpu(X,
-                                               output_class,
-                                               algo, task_category=1)
+        if predict_model == "CPU":
+            preds = self._predict_model_on_cpu(X, convert_dtype)
+
+        elif self.dtype == np.float64 and convert_dtype is False:
```

Comment:
Nitpick: `and not convert_dtype` would be more idiomatic (since it includes all "falsey" objects)